src/Tribe/Aggregator/Processes/Batch_Imports.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Class Tribe__Events__Aggregator__Processes__Batch_Imports
+ *
+ * @since TBD
+ */
+
+/**
+ * Class Tribe__Events__Aggregator__Processes__Batch_Imports
+ *
+ * Add custom hooks in order to support batch pushing.
+ *
+ * @since TBD
+ */
+class Tribe__Events__Aggregator__Processes__Batch_Imports {
+	/**
+	 * Update the endpoint used to initiate an process an import of events.
+	 *
+	 * @param string   $url      The completed URL generated.
+	 * @param string   $endpoint The path of the endpoint inside of the base url.
+	 * @param stdClass $api      An object representing the properties of the API.
+	 *
+	 * @return string The modified URL where to hit to process an import.
+	 */
+	public function build_url( $url, $endpoint, $api ) {
+		if ( 'import' !== $endpoint ) {
+			return $url;
+		}
+
+		return $api->domain . $api->path . 'v2.0.0' . '/' . $endpoint;
+	}
+
+	/**
+	 * Filter imports (if it has a parent import is a schedule import) and if the parent was not a batch pushing import,
+	 * make sure that that setting is respected, in this way we can support backwards compatibility as all imports created
+	 * before batch pushing are going to remaining using the old system and new imports are going to be considered as
+	 * batch pushing imports.
+	 *
+	 * @since TBD
+	 *
+	 * @param bool                                        $service_supports_batch_push If the current import has support for batch pushing.
+	 * @param Tribe__Events__Aggregator__Record__Abstract $abstract
+	 *
+	 * @return boolean If the current import supports batch pushing or not.
+	 */
+	public function allow_batch_import( $service_supports_batch_push, $abstract ) {
+		if ( ! $service_supports_batch_push ) {
```

Comment:
I'm a bit puzzled by the `$service_supports_batch_push` input argument.
If not supported, the method will return `false` out of the gate (refactor to return `false`).
The rest of the code keeps returning this input value while it should really return `true`: the rest. of the method would simply not be executed otherwise.

Reply:
This is mainly because this method is used in conjunction with a WordPress Filter where on a Filter the first argument is the current value on the filter. 

In this case as the value comes from the DB it can be a `falsy` value and not a strict`false` value by returning the original value from the filter we can ensure the filters chain can keep on using the current value on the filter only if specific conditions are meet.

Only `false` is returned in specif conditions, in this case if the import in place is a schedule import and the parent import (original import) was not created using Batch Pushing, this behavior would allow for a backwards compatible solution for customers with imports created before Batch Pushing so they can keep on using the old system to process imports.

Let me know in case you have additional questions.

Reply:
No further questions on this, thanks for clarifying.